packages/contracts-bedrock/src/L1/SuperchainConfig.sol

```diff
@@ -0,0 +1,280 @@
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    /// @notice Constructs the SuperchainConfig contract.
+    constructor() {
+        initialize({
+            _systemOwner: address(0),
+            _initiator: address(0),
+            _vetoer: address(0),
+            _guardian: address(0),
+            _delay: 0,
+            _sequencers: new T.SequencerKeys[](0)
+        });
+    }
+
+    /// @notice Initializer.
+    ///         The resource config must be set before the require check.
+    /// @param _systemOwner Owner of the contract.
+    /// @param _initiator   Address of the initiator who may initiate an upgrade or change to critical config values.
+    /// @param _vetoer      Address of the vetoer.
+    /// @param _guardian    Address of the guardian, can pause the OptimismPortal.
+    /// @param _delay       The delay time in seconds between when an upgrade is initiated and when it can be finalized.
+    /// @param _sequencers  The initial list of allowed sequencers
+    function initialize(
+        address _systemOwner,
+        address _initiator,
+        address _vetoer,
+        address _guardian,
+        uint256 _delay,
+        T.SequencerKeys[] memory _sequencers
+    )
+        public
+        reinitializer(2)
+    {
+        _setSystemOwner(_systemOwner);
+        _setInitiator(_initiator);
+        _setVetoer(_vetoer);
+        _setGuardian(_guardian);
+        _setDelay(_delay);
+
+        for (uint256 i = 0; i < _sequencers.length; i++) {
+            bytes32 sequencerHash = Hashing.hashSequencerKeys(_sequencers[i]);
+            allowedSequencers[sequencerHash] = true;
+            emit ConfigUpdate(UpdateType.ADD_SEQUENCER, abi.encode(_sequencers[i]));
+        }
+    }
+
+    /// @notice Returns an address stored in an arbitrary storage slot.
+    ///         These storage slots decouple the storage layout from
+    ///         solc's automation.
+    /// @param _slot The storage slot to retrieve the address from.
+    function _getAddress(bytes32 _slot) internal view returns (address addr_) {
+        assembly {
+            addr_ := sload(_slot)
+        }
+    }
+
+    /// @notice Stores an address in an arbitrary storage slot, `_slot`.
+    /// @param _slot The storage slot to store the address in.
+    /// @param _address The protocol version to store
+    /// @dev WARNING! This function must be used cautiously, as it allows for overwriting addresses
+    ///      in arbitrary storage slots.
+    function _setAddress(bytes32 _slot, address _address) internal {
+        assembly {
+            sstore(_slot, _address)
+        }
+    }
+
+    /// @notice Returns a uint256 stored in an arbitrary storage slot.
+    ///         These storage slots decouple the storage layout from
+    ///         solc's automation.
+    /// @param _slot The storage slot to retrieve the address from.
+    function _getValue(bytes32 _slot) internal view returns (uint256 value_) {
```

Comment:
Perhaps a follow up would be to turn these sorts of getters into a library because we have them defined in multiple contracts

Reply:
yeah, totally agree.